service/src/main/java/uk/gov/hmcts/reform/fpl/service/OthersService.java

```diff
@@ -39,4 +44,136 @@ private boolean otherExists(Others others) {
         return others != null && (others.getFirstOther() != null || others.getAdditionalOthers() != null);
     }
 
+    public Others modifyHiddenValues(Others others) {
+        final List<Element<Other>> othersList = new ArrayList<>();
+        Element<Other> firstOther;
+
+        if (others.getFirstOther().containsConfidentialDetails()) {
+            firstOther = Element.<Other>builder()
+                .value(others
+                    .getFirstOther()
+                    .toBuilder()
+                    .address(null)
+                    .telephone(null)
+                    .build())
+                .build();
+        } else {
+            firstOther = Element.<Other>builder()
+                .value(others.getFirstOther()).build();
+        }
+
+        others.getAdditionalOthers().stream().forEach(additionalOther -> {
+            if (additionalOther.getValue().containsConfidentialDetails()) {
+                othersList.add(Element.<Other>builder()
+                    .id(additionalOther.getId())
+                    .value(additionalOther.getValue().toBuilder().address(null).telephone(null).build())
+                    .build());
+            } else {
+                othersList.add(Element.<Other>builder()
+                    .id(additionalOther.getId())
+                    .value(additionalOther.getValue())
+                    .build());
+
+            }
+
+        });
+
+        return others.toBuilder().additionalOthers(othersList).firstOther(firstOther.getValue()).build();
+    }
+
+    public List<Element<Other>> modifyHiddenValuesConfidentialOthers(List<Element<Other>> confidentialOthers) {
+        final List<Element<Other>> confidentialOthersModified = new ArrayList<>();
+
+        confidentialOthers.stream().forEach(confidentialOther -> {
+            confidentialOthersModified.add(Element.<Other>builder()
+                .id(confidentialOther.getId())
+                .value(confidentialOther.getValue().toBuilder()
+                    .DOB(null)
+                    .gender(null)
+                    .birthPlace(null)
+                    .childInformation(null)
+                    .genderIdentification(null)
+                    .litigationIssues(null)
+                    .litigationIssuesDetails(null)
+                    .detailsHidden(null)
+                    .detailsHiddenReason(null)
+                    .build())
+                .build());
+        });
+
+        return confidentialOthersModified;
+    }
```

Comment:
what is this for?

Reply:
looks like it is called in the controller and then passed into addConfidentialDetailsToCaseDetails()